gen3_tracker/git/cli.py

```diff
@@ -0,0 +1,874 @@
+                    auth = gen3_tracker.config.ensure_auth(config=config)
+                results = ls(config, metadata={'project_id': config.gen3.project_id}, auth=auth)
+                object_ids = [_.object_id for _ in dvc_objects]
+            for _ in results['records']:
+                if _['did'] in object_ids:
+                    print('aws s3 cp ', _['urls'][0], _['file_name'])
+        else:
+            raise NotImplementedError(f"Remote {remote} not supported.")
+
+    except Exception as e:
+        click.secho(str(e), fg=ERROR_COLOR, file=sys.stderr)
+        if config.debug:
+            raise
+
+
+@cli.command()
+@click.argument('project_id', default=None, required=False, envvar=f"{gen3_tracker.ENV_VARIABLE_PREFIX}PROJECT_ID", metavar='PROJECT_ID')
+@click.pass_obj
+def clone(config, project_id):
+    """Clone a repository into a new directory"""
+    try:
+        config.gen3.project_id = project_id
+        assert not pathlib.Path(project_id).exists(), f"{project_id} already exists.  Please remove it first."
+        os.mkdir(project_id)
+        os.chdir(project_id)
+        with Halo(text='Cloning', spinner='line', placement='right', color='white'):
+            auth = gen3_tracker.config.ensure_auth(config=config)
+            snapshot, zip_filepath = download_snapshot(auth, config)
+            assert not pathlib.Path('.git').exists(), "A git repository already exists.  Please remove it, or move to another directory first."
+            # unzip
+            with zipfile.ZipFile(zip_filepath, 'r') as zip_ref:
+                zip_ref.extractall('.')
+
+            # if we just unzipped a .git these directories will exist
+            expected_dirs = ['.git', 'META', 'MANIFEST']
+            if not all([pathlib.Path(_).exists() for _ in expected_dirs]):
+                # if not, we have downloaded a legacy SNAPSHOT.zip, so lets migrate the data to the expected drirectories
+                click.secho(f"{expected_dirs} not found after downloading {snapshot['file_name']} processing legacy snapshot", fg=INFO_COLOR, file=sys.stderr)
+                # legacy - was this a *SNAPSHOT.zip?
+                meta_files = (pathlib.Path('studies') / config.gen3.project)
+                # legacy - was this a *meta.zip?
+                if not meta_files.exists():
+                    meta_files = pathlib.Path('.')
+                # create local directories and git
+                [_ for _ in config_init(config, project_id)]
+                ensure_git_repo(config=config)
+                # move ndjson from studies to META
+                for _ in meta_files.glob('*.ndjson'):
+                    shutil.move(_, 'META/')
+                # add to git
+                run_command('git add META/*.*')
+                # migrate DocumentReferences to MANIFEST
+                references = meta_index()
+                manifest_files = []
+                for _ in read_ndjson_file('META/DocumentReference.ndjson'):
```

Comment:
Unsure about this `read_ndjson_file('META/DocumentReference.ndjson')` pattern. Getting DVC information from the fhir record makes sense, but What if the actual md5, size, name of the actual file aren't the same as what the fhir record says? Is there a check for this?